app/src/main/java/com/zulip/android/ZulipApp.java

```diff
@@ -261,4 +264,29 @@ public void markMessageAsRead(Message message) {
             unreadMessageHandler.sendEmptyMessageDelayed(0, 2000);
         }
     }
+    public void MuteTopic(int streamId, String subject) {
+        System.out.println((streamId + "-" + subject));
+        muted_topics.add(streamId + "-" + subject);
+        SharedPreferences.Editor editor = settings.edit();
+        editor.putStringSet("muted_topics", muted_topics);
+        editor.commit();        
+    }
+
+    public void UnMuteTopic(int streamId, String subject) {
+        muted_topics.remove(streamId + "-" + subject);
+        SharedPreferences.Editor editor = settings.edit();
+        editor.putStringSet("muted_topics", muted_topics);
```

Comment:
`putStringSet` is buggy when used in this manner.  See http://developer.android.com/reference/android/content/SharedPreferences.html#getStringSet%28java.lang.String,%20java.util.Set%3Cjava.lang.String%3E%29.  You'll need to either create a new object to pass in to save, or clear out the value before saving.